lldb/source/Expression/IRMemoryMap.cpp

```diff
@@ -640,6 +640,9 @@ void IRMemoryMap::WritePointerToMemory(lldb::addr_t process_address,
                                        lldb::addr_t address, Status &error) {
   error.Clear();
 
+  if (auto process_sp = GetProcessWP().lock())
+    address = process_sp->FixAnyAddress(address);
```

Comment:
Why not put this deeper down, in WriteScalarToMemory or even WriteMemory ?

Reply:
Only pointers need to be fixed though. I'm not sure I understand?

Reply:
I will make a note of that possibility and try to make things break when I get the chance, as this is the kind of corner case I want to explore too.

Reply:
I think the crucial part of your observations is that we really need to delay cleaning those attributes until absolutely necessary (e.g. https://github.com/llvm/llvm-project/pull/150537), so that we then have the chance to decide whether/which bits to clean up. In that sense, the existing patches are a step in that direction.